.github/workflows/push.yml

```diff
@@ -0,0 +1,57 @@
+name: Single Build & Test (Push)
+
+# workflow is for a branch push only and ignores master.
+# push to master (which is also pull requet merge) has a more elaborate workflow to run
+# github repo is configured with branch protection on master.
+on: 
+  push:
+    branches-ignore:
+      - 'master'
+
+  workflow_dispatch:
+    inputs: 
+      node:
+        required: false
+        description: 'A node version (e.g 10.16.0)'
+        default: '14' # TODO: update to 16 when support is added
+
+jobs:
+  build-single-test:
+    runs-on: ubuntu-latest # environment job will run in
+
+    env:
+      # TODO: move to staging when AO-20147
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }} 
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Setup Node ${{ github.event.inputs.node || github.event.inputs.node.default }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ github.event.inputs.node || github.event.inputs.node.default }}
```

Comment:
Same as above comment, in the event that this workflow was triggered through a push, the node-version will be undefined, hence the default value will now suddenly be determined by the setup-node action.

Reply:
It does work on push (https://github.com/appoptics/appoptics-bindings-node/actions/runs/1016062848) *because* of this default behavior. Default input is 14. When none, default node-version is false, thus ignored. I agree it is a little cheeky... early morning tweaks should be avoided. Will replace `github.event.inputs.node.default` with `'14'`.